src/core/service.c

```diff
@@ -1945,10 +1945,18 @@ static void service_enter_restart(Service *s) {
 }
 
 static void service_enter_reload_by_notify(Service *s) {
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        int r;
+
         assert(s);
 
         service_arm_timer(s, usec_add(now(CLOCK_MONOTONIC), s->timeout_start_usec));
         service_set_state(s, SERVICE_RELOAD);
+
+        /* service_enter_reload_by_notify is never called during a reload, thus no loops are possible. */
+        r = manager_propagate_reload(UNIT(s)->manager, UNIT(s), JOB_FAIL, &error);
```

Comment:
Would this effectively reload the dependant units when `RELOADING=1` is sent?

Shouldn't this be done when the unit goes back to running state (after the `READY=1` notification)?